### PR TITLE
allows specifying components to include in scheduler state

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- allows specifying components to include in scheduler state

## Why are we making these changes?

Reduces the amount of scheduler state that is transmitted and rendered when only certain components are of interest.

### CODE REVIEW

File: waiter/src/waiter/scheduler/composite.clj [hunk 1]
```diff
@@ -99,9 +99,13 @@
         service-id->scheduler
         (scheduler/service-id->state service-id)))
 
-  (state [_]
-    {:aggregator (query-aggregator-state-fn)
-     :components (pc/map-vals scheduler/state scheduler-id->scheduler)})
+  (state [_ include-flags]
+    (cond-> {:supported-include-params ["aggregator" "components"]
+             :type "CompositeScheduler"}
```
Comment (DaoWen) [inline waiter/src/waiter/scheduler/composite.clj:104]: Isn't it redundant to include the word "Scheduler" in each of these scheduler types?

Comment (shamsimam) [reply waiter/src/waiter/scheduler/composite.clj:104]: Dropped the `Scheduler` portion from the types

File: waiter/src/waiter/scheduler/shell.clj [hunk 2]
```diff
@@ -686,10 +686,15 @@
...
-    {:id->service @id->service-agent
-     :port->reservation @port->reservation-atom
-     :syncer (retrieve-syncer-state-fn)})
+  (state [_ include-flags]
+    (cond-> {:supported-include-params ["port-reservation" "services" "syncer"]
+             :type "ShellScheduler"}
+      (contains? include-flags "port-reservation")
+      (assoc :port->reservation @port->reservation-atom)
+      (contains? include-flags "services")
+      (assoc :id->service @id->service-agent)
```
Comment (DaoWen) [inline waiter/src/waiter/scheduler/shell.clj:695]: I think it's very confusing that the "include" flag doesn't match the key here. (And this isn't the only one.)

Is there a reason you're not just using the key name as the flag name?

Comment (shamsimam) [reply waiter/src/waiter/scheduler/shell.clj:695]: The key name is very wordy, so used a shorter variant. The include flags are passed in the response so users don't have to guess what flags are supported.

Comment (shamsimam) [reply waiter/src/waiter/scheduler/shell.clj:693]: Changed the names to match the keys

Comment (sradack) [reply waiter/src/waiter/scheduler/shell.clj:693]: Nick looks like he covered this.

File: waiter/src/waiter/scheduler/composite.clj (hunk 1 above)
Comment (sradack) [reply waiter/src/waiter/scheduler/composite.clj]: Nick looks like he covered this.

Comment (sradack) [reply waiter/src/waiter/scheduler/composite.clj]: I liked it how it was before.  The type name is supposed to match the actual type name, yes?

Comment (shamsimam) [reply waiter/src/waiter/scheduler/composite.clj]: I'm happy to revert it :)

Comment (shamsimam) [reply waiter/src/waiter/scheduler/composite.clj:104]: Reverted to include `Scheduler` suffix.

Comment (DaoWen) [reply waiter/src/waiter/scheduler/composite.clj:104]: I still disagree that it's useful if _every possible value_ of this string is _always_ going to end in "Scheduler", but this is just an aesthetic thing so I'm not going to push it. I don't anticipate ever using this field for anything other than visually confirming that I'm looking at the state that I expected.

**Pull request merged**